src/item_uid.h

```diff
@@ -0,0 +1,42 @@
+#ifndef ITEM_UID_H
+#define ITEM_UID_H
+
+#include <limits>
+
+#include "json.h"
+
+class item_uid : public JsonSerializer, public JsonDeserializer
```

Comment:
See https://github.com/CleverRaven/Cataclysm-DDA/pull/15682#discussion_r55389851 for some issues I raised with a previous proposal for a uid.  if it's guaranteed to be a 64-bit number, we actually do have enough numbers to keep handing them out indefinitely, but it needs to be versioned so we can transition them to new schemas if necessary, and it needs to have a player id injected into it to handle games being played concurrently in the same game world.

Also please make a uid factory that's a member of game instead of putting the uid logic in the game class.  In fact a uid factory is called for but I'm not sure a uid type is, it's just a big number, with the schema encoded in the value of the number.